### DISCUSSION TITLE
Intersection line and grid

### DISCUSSION THREAD

**Discussion by issimo007** [pyvista/pyvista#3326] [Q&A] [answered]
Hello is there a way to calculate the intersection of a line (defined in terms of XYZ) and a surface (defined by a grid, in ZMAP format or other XYZ)?  If so, does the intersection XYZ accuracy depends on the surface grid resolution or does PyVista make some sort of interpolation between the grid points?  Thank you!

**Comment by banesullivan** [comment, marked as answer]
The ray tracing example is a great place to start: 

- https://docs.pyvista.org/examples/99-advanced/ray-trace.html
- https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyDataFilters.ray_trace.html

See also:

- https://docs.pyvista.org/user-guide/optional_features.html?highlight=ray+tracing#vectorised-ray-tracing
- https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyData.intersection.html?highlight=intersect

> does the intersection XYZ accuracy depends on the surface grid resolution or does PyVista make some sort of interpolation between the grid points?

If you use the `ray_trace` filter, the accuracy will depend on the resolution of the mesh being intersected. The ray trace will give you the exact coordinates on the face of the cell that is intersected in the mesh

**Reply by issimo007** [reply]
Thank you!  I really appreciate the tips.
